tests/device_tests/bitcoin/test_signtx.py

```diff
@@ -738,77 +705,81 @@ def test_p2sh(client):
         )
 
-    assert (
-        tx_hash(serialized_tx).hex()
-        == "5042aed319b9f018d693dbf8f3db926ee4ab4dae670a2911625b440a1366f79d"
+    assert_tx_matches(
+        serialized_tx,
+        hash_link="https://tbtc1.trezor.io/api/tx/525693745ef44161ed32f3def7f13e92d96504e06448910ce2e945bfe5e8cc7b",
+        tx_hex="01000000018eacb9dc2bfa59e34b57ed9c49a14b4e783ad77fc8e43e5483cf25135d6ad558000000006a473044022029b7d07f068501dc7a5dcf5148167a286a949483cac88b6f85b3ae92baa3346902203488709587453467248d52d0f18fa36d43909854ee1958feef5c9b78c509d15d012103f5008445568548bd745a3dedccc6048969436bf1a49411f60938ff1938941f14ffffffff01409c00000000000017a9147f844bdb0b8fd54b64e3d16c85dc1170f1ff97c18700000000",
     )
 
 
 def test_testnet_big_amount(client):
     # This test is testing transaction with amount bigger than fits to uint32
 
-    # tx: 2bac7ad1dec654579a71ea9555463f63ac7b7df9d8ba67b4682bba4e514d0f0c:1
-    # input 1: 411102528330 Satoshi
+    # input tx: 074b0070939db4c2635c1bef0c8e68412ccc8d3c8782137547c7a2bbde073fc0:1
 
     inp1 = messages.TxInputType(
-        address_n=parse_path("m/44'/1'/0'/0/0"),
-        amount=411102528330,
-        prev_hash=TXHASH_2bac7a,
+        address_n=parse_path("44h/1h/0h/0/6"),  # mnY26FLTzfC94mDoUcyDJh1GVE3LuAUMbs
+        amount=4_500_000_000,
+        prev_hash=TXHASH_074b00,
         prev_index=1,
     )
     out1 = messages.TxOutputType(
-        address="mopZWqZZyQc3F2Sy33cvDtJchSAMsnLi7b",  # seed allallall, bip32: m/44'/1'/0'/0/1
-        amount=411102528330,
+        address="2N5daLhptwpXPBY84TQ2AjeLLkL8ru7n6ai",  # top secret address
+        amount=4_500_000_000 - 10_000,
         script_type=messages.OutputScriptType.PAYTOADDRESS,
     )
     _, serialized_tx = btc.sign_tx(
         client, "Testnet", [inp1], [out1], prev_txes=TX_CACHE_TESTNET
     )
-    assert (
-        serialized_tx.hex()
-        == "01000000010c0f4d514eba2b68b467bad8f97d7bac633f465595ea719a5754c6ded17aac2b010000006b4830450221008e3b926f04d8830bd5b67698af25c9e00c9db1b1ef3e5d69af794446753da94a02202d4a7509f26bba29ff643a7ac0d43fb128c1a632cc502b8f44eada8930fb9c9b0121030e669acac1f280d1ddf441cd2ba5e97417bf2689e4bbec86df4f831bf9f7ffd0ffffffff014ac39eb75f0000001976a9145b157a678a10021243307e4bb58f36375aa80e1088ac00000000"
+
+    assert_tx_matches(
+        serialized_tx,
+        hash_link="https://tbtc1.trezor.io/api/tx/890155c89905c86fb087eddbe9973fa845c99913836e872f41fbdfa3864b607d",
+        tx_hex="0100000001c03f07debba2c747751382873c8dcc2c41688e0cef1b5c63c2b49d9370004b07010000006b483045022100a2c79eaed632746fd514aa09eae51c0294bdf66c74619306e0273cd1a470c9e7022050c78dec6acd65b42150dba70e2b546dfd737d77f46975a97427999cb2b8280401210344e14b3da8f5fe77a5465d0f8fe089d64ed5517d1f1f989edd00f530938a2c22ffffffff01f065380c0100000017a91487dba64df7e9386d0b0f3ef557269833e12d1b7a8700000000",
     )
 
 
-@pytest.mark.setup_client(mnemonic=MNEMONIC12)
 def test_attack_change_outputs(client):
```

Comment:
this test should really fail when it doesn't trigger change detection.

please add `expected_responses` here, before the attack processor is installed.

i believe you can use the same data as in `test_two_two`, and the same expected_responses?

Reply:
Addressed in eaaa13742bdbe50f06a8452d8f6a93a52365a211 - taking the `test_two_two` details was alright